app/models/request.rb

```diff
@@ -80,12 +80,20 @@ def find_existing_user
     if user.webauth_user?
       User.find_by_webauth(user.webauth)
     elsif user.library_id_user?
-      User.find_by_library_id(user.library_id)
+      find_exisisting_library_id_user
     elsif user.non_webauth_user?
       find_existing_email_user
     end
   end
 
+  def find_exisisting_library_id_user
+    if user.email
+      User.find_by('library_id = ? AND email = ?', user.library_id, user.email)
```

Comment:
Active record provides methods to do this type of thing natively.

`User.where(library_id: user.library_id, email: user.email).first`